src/sinks/util/http.rs

```diff
@@ -127,11 +128,17 @@ impl RetryLogic for HttpRetryLogic {
         error.is_connect() || error.is_closed()
     }
 
-    fn should_retry_response(&self, response: &Self::Response) -> bool {
+    fn should_retry_response(&self, response: &Self::Response) -> Option<Cow<str>> {
```

Comment:
I wonder if this PR makes sense to hold off on until we get the error types merged? I'd much rather see this return type look like `Option<&dyn crate::Error>`.

Reply:
Interesting idea. Yes this could be another application of that type.

Reply:
Ping on this, what do you think now that you've done a bit more error stuff? I personally don't really like exposing Cow<str> like this honestly. But I am fine either way.

Reply:
Obviously, I can't use the `vector::Error` type until the other PR gets merged. However...

I think this is a similar pattern, but it's not quite an `Error` the same way the other uses would be. An enumerated return would be preferable here. I have no strong opinions between an enum specific to this function (and shared across all users) or a boxed type like `Error`.

I chose `Cow<str>` because some of the return values are just static strings. This is probably not a hot path in any normal operation (we're already going to retry) so I was probably over-engineering again.

Reply:
Right, my only gripe is just this seems to complicate but that may just be my personal view 😅